From d9dc30e089f6d65caa85429347a0ff898ba74648 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 4 Nov 2022 14:22:25 +0100
Subject: [PATCH 0327/1695] dts: stm32: add default HCONF1 value for VDDIO on
 stm32mp257f-ev1 board

Add the default initialization values for stm32mp25 EV1 board.
Update the shadow default value to allow the system
to freeze the watchdog in stop or standby.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 6cdd65446381d037ed1e383d1e59a88885bc89ff dts: stm32: add default HCONF1 value for VDDIO on stm32mp257f-ev1 board]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/302934
Change-Id: I7747d93e34438cad3325df76f3eb459a90b38253
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349481
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-ev1.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 897b593da..c1e4d29f1 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -7,6 +7,7 @@
 /dts-v1/;
 
 #include <dt-bindings/clock/stm32mp25-clksrc.h>
+#include <dt-bindings/soc/stm32mp-provisioning.h>
 #include <dt-bindings/soc/stm32mp25-rif.h>
 #include <dt-bindings/soc/stm32mp25-risab.h>
 #include <dt-bindings/soc/stm32mp25-risaf.h>
@@ -35,6 +36,14 @@
 		device_type = "memory";
 		reg = <0x0 0x80000000 0x1 0x00000000>;
 	};
+
+	shadow-prov {
+		compatible = "st,provisioning";
+		hconf1_prov {
+			nvmem-cells = <&hconf1_otp>;
+			st,shadow-value = <0x00018DB6>;
+		};
+	};
 };
 
 &risaf2 {
-- 
2.47.2

